### PR TITLE
Simplify `cattrs._compat.is_typeddict`

### PR DESCRIPTION
**Pull request by AlexWaygood**
Following up from https://github.com/python/typing_extensions/issues/230#issuecomment-1587750089! Not sure why the test for generic typeddicts on 3.11 is failing.

This:

- Is simpler
- Is more robust (it doesn't involve importing private implementation details from typing/typing_extensions, which are liable to be changed without warning)
- Doesn't add any additional dependency on typing_extensions except on Python <3.10

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Hmm, the test suite is succeeding on PyPy but the "upload to codecov" bit is failing, and then that leads to all other jobs being cancelled

File: src/cattrs/_compat.py [hunk 1]
```diff
@@ -20,15 +20,18 @@
 from attr import fields as attrs_fields
 from attr import resolve_types
 
+__all__ = ["ExtensionsTypedDict", "is_typeddict", "TypedDict"]
```
[the remark this answers is not in the document]
Comment (layday) [reply src/cattrs/_compat.py:23]: whoops, my bad. You might wanna add `ExceptionGroup` too which uses the `x as x` syntax currently.

**Comment by Tinche** [comment]
Wow Codecov is really being garbage today, sorry about that heh.

**Comment by Tinche** [comment]
Here's a reduced version of the failing test case:

```python
from typing import TypedDict, Generic, TypeVar
from cattrs._compat import is_typeddict

T = TypeVar("T")

class A(TypedDict, Generic[T], total=True):
    a: T

print(is_typeddict(A[int]))  # False, should be true
```

**Comment by AlexWaygood** [comment]
> Here's a reduced version of the failing test case:

Ahh, thank you! Should be fixed now -- all tests are now passing for me locally :)

**Comment by AlexWaygood** [comment]
Also, I should have spotted that behaviour difference between cattrs's version of `is_typeddict` and the typing(_extensions) one just by looking at the code 🤦

**Comment by Tinche** [comment]
LGTM, <3 can you just add a line to the changelog? Doesn't have to be very specific, but might help someone down the line if there's a behavior change we're not testing for.

**Comment by AlexWaygood** [comment]
> LGTM, <3 can you just add a line to the changelog? Doesn't have to be very specific, but might help someone down the line if there's a behavior change we're not testing for.

Done! Lmk if that's not quite what you were looking for :)

**Review by Tinche — APPROVED** [review]

**Comment by Tinche** [comment]
Great, thanks!

**Pull request merged**